synapse/handlers/register.py

```diff
@@ -220,8 +220,26 @@ def register(
 
         # auto-join the user to any rooms we're supposed to dump them into
         fake_requester = create_requester(user_id)
+
+        # try to create the room if we're the first user on the server
+        if self.hs.config.autocreate_auto_join_rooms:
+            count = yield self.store.count_all_users()
+            auto_create_rooms = count == 1
```

Comment:
you need to initialise `auto_create_rooms` for the case where `autocreate_auto_join_rooms` is False.